### PR TITLE
Use Cassandra driver v4

### PR DESCRIPTION
**Pull request by findinpath**
Replace the legacy Cassandra client driver

`com.datastax.cassandra:cassandra-driver-core:3.x`

with

`com.datastax.oss:java-driver-core:4.x`

### CODE REVIEW

**Comment by findinpath** [comment]
@ebyhr , @wendigo FYI

File: tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java [hunk 1]
```diff
@@ -77,87 +74,73 @@
 
     public CassandraQueryExecutor(Configuration configuration)
     {
-        cluster = Cluster.builder()
-                .addContactPoint(configuration.getStringMandatory("databases.cassandra.host"))
-                .withPort(configuration.getIntMandatory("databases.cassandra.port"))
+        session = CqlSession.builder()
+                .addContactPoint(new InetSocketAddress(configuration.getStringMandatory("databases.cassandra.host"), configuration.getIntMandatory("databases.cassandra.port")))
+                .withLocalDatacenter("datacenter1")
```
Comment (ebyhr) [inline tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java]: I think we shouldn't hard-code this value. How about retrieving from configuration?

File: tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java [hunk 2]
```diff
@@ -77,87 +74,73 @@
...
     public List<String> getColumnNames(String keySpace, String tableName)
     {
-        checkState(tableExists(keySpace, tableName), "table %s.%s does not exist", keySpace, tableName);
-        KeyspaceMetadata keyspaceMetadata = session.getCluster().getMetadata().getKeyspace(keySpace);
-        TableMetadata tableMetadata = keyspaceMetadata.getTable(tableName);
-        return tableMetadata.getColumns().stream().map(ColumnMetadata::getName).collect(toList());
+        Optional<KeyspaceMetadata> keyspaceMetadata = session.getMetadata().getKeyspace(keySpace);
+        if (!keyspaceMetadata.isPresent()){
+            throw new IllegalArgumentException(format("keyspace %s does not exist", keySpace));
+        }
```
Comment (ebyhr) [inline tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java]: nit: Let's simplify these code:
```java
        KeyspaceMetadata keyspaceMetadata = session.getMetadata().getKeyspace(keySpace)
                .orElseThrow(() -> new IllegalArgumentException(format("keyspace %s does not exist", keySpace)));
```

**Review by ebyhr — APPROVED** [review]

File: tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java (hunk 1 above)
Comment (findinpath) [reply tempto-core/src/main/java/io/trino/tempto/internal/query/CassandraQueryExecutor.java]: Good point. I introduced the optional property `databases.cassandra.local_datacenter` for this purpose.

**Pull request merged**